build-machinery/make/targets/openshift/codegen.mk

```diff
@@ -0,0 +1,34 @@
+CODEGEN_PKG ?=./vendor/k8s.io/code-generator/
+CODEGEN_GENERATORS ?=all
+CODEGEN_OUTPUT_BASE ?=../../..
+CODEGEN_GO_HEADER_FILE ?=/dev/null
+
+#ifneq ($(and $(CODEGEN_GENERATORS),$(CODEGEN_OUTPUT_PACKAGE),$(CODEGEN_API_PACKAGE),$(CODEGEN_GROUPS_VERSION)),)
+#	$(error Some of required CODEGEN_* variables is unset!)
+#endif
+define run-codegen
+$(CODEGEN_PKG)/generate-groups.sh \
+	"$(CODEGEN_GENERATORS)" \
+	"$(CODEGEN_OUTPUT_PACKAGE)" \
+	"$(CODEGEN_API_PACKAGE)" \
+	"$(CODEGEN_GROUPS_VERSION)" \
+    --output-base $(CODEGEN_OUTPUT_BASE) \
+    --go-header-file $(CODEGEN_GO_HEADER_FILE) \
+    $1
+endef
+
+
+verify-codegen:
+	$(call run-codegen,--verify-only)
+.PHONY: verify-codegen
```

Comment:
some other phony targets are missing in other files. E.g. "help"

Reply:
I missed `help`, are there others?

Reply:
fixed wherever I saw it